### PR TITLE
Move UA exceptions to remote config

### PR DESCRIPTION
**Pull request by marcosholgado**
Task/Issue URL: https://app.asana.com/0/1125189844152671/1202317386594677

### Description
This PR moves the hardcoded UA exception lists to the remote config.

### Steps to test this PR
Before you begin, using Charles or similar map the request to `https://staticcdn.duckduckgo.com/trackerblocking/config/v2/android-config.json` to return the json attached in the Asana task.

Our UA looks like `Mozilla/5.0 (Linux; Android 12) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/101.0.4951.61 Mobile DuckDuckGo/5 Safari/537.36`

Notice the application component: `DuckDuckGo/5`
and the version component: `Version/4.0`

_User Agent works as expected_
- [x] Install from this branch
- [x] In the logcat filter by useragent
- [x] Go to SERP
- [x] In the logcat you should see contain the version and application component
- [x] Go to `cvs.com`
- [x] In the logcat the UA should not contain the application component and include the version component
- [x] Go to `ing.nl` 
- [x] In the logcat the UA should not contain the version component and include the application component
- [x] Go to `marcos.com`
- [x] In the logcat the UA should not contain both the version component and the application component
- [x] Go to `facebook.com`
- [x] In the logcat the UA should be the desktop version `(X11; Linux aarch64)` instead of `(Linux; Android 12)`
- [x] The UA should also not contain the version component. The UA change will probably show twice, this is normal.
- [x] Go to `cnn.com`
- [x] The UA should be the default one from WebView, no application or version component should show up and extra information that we remove should be re-added.
- [x] Go to `chase.com`
- [x] The UA should be the default one from WebView, no application or version component should show up and extra information that we remove should be re-added.
- [x] Go to `sovietgames.ru`
- [x] The UA should be the default one from WebView, no application or version component should show up and extra information that we remove should be re-added.
- [x] Go to `cvs.com` and remove privacy protections. The website should reload.
- [x] The UA should be the default one from WebView, no application or version component should show up and extra information that we remove should be re-added.

_Disable User Agent_
- [x] Using Flipper or similar disable the UA toggle in shared preferences (or change the json to "disabled" and install the app again)
- [x] Reload the app
- [x] Go to SERP
- [x] The UA should be the default one from WebView, no application or version component should show up and extra information that we remove should be re-added.
- [x] Go to the options menu and select `Desktop Site`
- [x] The UA should be the default one from WebView but should contain the desktop portion instead of the mobile one `(X11; Linux aarch64)` instead of `(Linux; Android 12)`.

### CODE REVIEW

File: privacy-config/privacy-config-impl/src/main/java/com/duckduckgo/privacy/config/impl/features/useragent/RealUserAgent.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.privacy.config.api.UserAgent
+import com.duckduckgo.privacy.config.impl.features.unprotectedtemporary.UnprotectedTemporary
+import com.duckduckgo.privacy.config.store.features.useragent.UserAgentRepository
+import com.squareup.anvil.annotations.ContributesBinding
+import dagger.SingleInstanceIn
+import javax.inject.Inject
+
+@ContributesBinding(AppScope::class)
+@SingleInstanceIn(AppScope::class)
```
Comment (aitorvs) [inline privacy-config/privacy-config-impl/src/main/java/com/duckduckgo/privacy/config/impl/features/useragent/RealUserAgent.kt:29]: nit: singleton prob not needed as it doesn't hold state

**Review by aitorvs — APPROVED** [review]
All works as expected.

Left couple nits about (not) singletons but feel free to ignore them. No strong opinions

File: app/src/main/java/com/duckduckgo/app/privacy/db/UserAllowListRepository.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import dagger.SingleInstanceIn
+import javax.inject.Inject
+
+interface UserAllowListRepository {
+    fun isDomainInUserAllowList(domain: String): Boolean
+}
+
+@SingleInstanceIn(AppScope::class)
```
[the remark this answers is not in the document]
Comment (marcosholgado) [reply app/src/main/java/com/duckduckgo/app/privacy/db/UserAllowListRepository.kt]: 👍

File: privacy-config/privacy-config-impl/src/main/java/com/duckduckgo/privacy/config/impl/features/useragent/RealUserAgent.kt (hunk 1 above)
Comment (marcosholgado) [reply privacy-config/privacy-config-impl/src/main/java/com/duckduckgo/privacy/config/impl/features/useragent/RealUserAgent.kt]: 👍

**Pull request merged**